### PR TITLE
[SASS-2414] migrate view content tests from integration to unit

### PR DESCRIPTION
**Pull request by rtna-hmrc**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-2414](https://jira.tools.tax.service.gov.uk/browse/SASS-2414)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 1]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              taxablePayToDate = Some(3000.00),
+              totalTaxToDate = Some(300.00),
+              currentDataIsHmrcHeld = false
+            ),
+            studentLoans = Some(StudentLoansCYAModel(
+              uglDeduction = false, uglDeductionAmount = None, pglDeduction = true, pglDeductionAmount = None))
+          )))
+        userDataStub(IncomeTaxUserData(), nino, taxYearEOY)
+
+        urlGet(url(taxYearEOY), false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:84]: The welsh content parameter boolean is defaulted to 'false' and so can be removed.

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 2]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              taxablePayToDate = Some(3000.00),
+              totalTaxToDate = Some(300.00),
+              currentDataIsHmrcHeld = false
+            ),
+            studentLoans = Some(StudentLoansCYAModel(
+              uglDeduction = false, uglDeductionAmount = None, pglDeduction = true, pglDeductionAmount = Some(100.00)))
+          )))
+        userDataStub(IncomeTaxUserData(), nino, taxYearEOY)
+
+        urlGet(url(taxYearEOY), false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:115]: Same as above

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 3]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              totalTaxToDate = Some(300.00),
+              currentDataIsHmrcHeld = false
+            ),
+            studentLoans = Some(StudentLoansCYAModel(
+              uglDeduction = true, Some(1000.22), pglDeduction = true, Some(3000.22)
+            ))
+          )))
+        userDataStub(IncomeTaxUserData(), nino, taxYearEOY)
+
+        urlGet(url(taxYearEOY), false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:147]: Same as above

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 4]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+                uglDeduction = true, Some(2000.00), pglDeduction = true, Some(4000.00)
+              ))
+            )
+          ))
+
+          userDataStub(incomeTaxUserData, nino, taxYearEOY)
+          urlPost(
+            url(taxYearEOY),
+            body = Map("amount" -> pglDeductionAmount.toString),
+            false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:246]: Same as above

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 5]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              taxablePayToDate = Some(3000.00),
+              totalTaxToDate = Some(300.00),
+              currentDataIsHmrcHeld = false
             )
-          }
+          )))
+        userDataStub(IncomeTaxUserData(), nino, taxYearEOY)
 
-          implicit val document: () => Document = () => Jsoup.parse(result.body)
+        urlGet(url(taxYearEOY), follow = false, welsh = false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:176]: Same as above

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 6]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+    "redirect to student loans cya page when there is no employment user data returned" in {
+      lazy val result = {
+        dropEmploymentDB()
+        authoriseAgentOrIndividual(false)
 
-          "result should be BadRequest" in {
-            result.status shouldBe BAD_REQUEST
-          }
-        }
+        urlGet(url(taxYearEOY), follow = false, welsh = false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:189]: Same as above

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 7]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              uglDeduction = true, Some(2000.00), pglDeduction = true, Some(4000.00)
+            ))
+          )
+        ))
+
+        userDataStub(incomeTaxUserData, nino, taxYearEOY)
+        urlPost(
+          url(taxYearEOY),
+          body = Map("amount" -> ""),
+          false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:286]: Same as above

File: it/controllers/studentLoans/PglAmountControllerISpec.scala [hunk 8]
```diff
@@ -145,366 +56,297 @@ class PglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              uglDeduction = true, Some(2000.00), pglDeduction = true, Some(4000.00)
+            ))
+          )
+        ))
+
+        userDataStub(incomeTaxUserData, nino, taxYearEOY)
+        urlPost(
+          url(taxYearEOY),
+          body = Map("amount" -> "abc"),
+          false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/PglAmountControllerISpec.scala:327]: Same as above

File: it/controllers/studentLoans/UglAmountControllerISpec.scala [hunk 9]
```diff
@@ -151,419 +91,164 @@ class UglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+              didYouLeaveQuestion = Some(false),
+              taxablePayToDate = Some(3000.00),
+              totalTaxToDate = Some(300.00),
+              currentDataIsHmrcHeld = false
+            )
+          )))
+        userDataStub(IncomeTaxUserData(), nino, taxYearEOY)
 
-        }
+        urlGet(url(taxYearEOY), follow = false, welsh = false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/UglAmountControllerISpec.scala:118]: The welsh content parameter boolean is defaulted to 'false' and so can be removed.

File: it/controllers/studentLoans/UglAmountControllerISpec.scala [hunk 10]
```diff
@@ -151,419 +91,164 @@ class UglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
-              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
-          }
+      lazy val result = {
+        dropEmploymentDB()
+        authoriseAgentOrIndividual(false)
 
-          result.status shouldBe SEE_OTHER
-          result.headers("Location").headOption shouldBe Some(controllers.studentLoans.routes.StudentLoansCYAController.show(taxYearEOY, employmentId).url)
-        }
+        urlGet(url(taxYearEOY), follow = false, welsh = false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/UglAmountControllerISpec.scala:133]: Same as above

File: it/controllers/studentLoans/UglAmountControllerISpec.scala [hunk 11]
```diff
@@ -151,419 +91,164 @@ class UglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
-          "result should be BadRequest" in {
-            result.status shouldBe BAD_REQUEST
-          }
+          ))
+
+          userDataStub(incomeTaxUserData, nino, taxYearEOY)
+          urlPost(
+            url(taxYearEOY),
+            body = Map("amount" -> "1234"),
+            false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/UglAmountControllerISpec.scala:183]: Same as above

File: it/controllers/studentLoans/UglAmountControllerISpec.scala [hunk 12]
```diff
@@ -151,419 +91,164 @@ class UglAmountControllerISpec extends IntegrationTest with ViewHelpers with Emp
...
+                uglDeduction = true, Some(20000), pglDeduction = true, None
+              ))
+            )
+          ))
+
+          userDataStub(incomeTaxUserData, nino, taxYearEOY)
+          urlPost(
+            url(taxYearEOY),
+            body = Map("amount" -> "1234"),
+            welsh = false,
```
Comment (dnyuar) [inline it/controllers/studentLoans/UglAmountControllerISpec.scala:225]: Same as above

**Review by dnyuar — CHANGES_REQUESTED** [review]

File: it/controllers/studentLoans/PglAmountControllerISpec.scala (hunk 1 above)
Comment (rtna-hmrc) [reply it/controllers/studentLoans/PglAmountControllerISpec.scala]: Done for all of the occurrences

**Pull request merged**